src/components/accordion/Accordion.jsx

```diff
@@ -161,25 +242,71 @@ const Accordion = ({
     }
   }
 
+  const [prevIndex, setPrevIndex] = useState();
+
+  if (isControlled) {
```

Comment:
what about combining the condition to reduce nesting?  
```suggestion
  if (isControlled && expanded !== prevIndex) {
```